shell/platform/android/platform_view_android_jni_impl.cc

```diff
@@ -1141,10 +1141,9 @@ void PlatformViewAndroidJNIImpl::FlutterViewHandlePlatformMessageResponse(
                         nullptr);
   } else {
     // Convert the vector to a Java byte array.
-    fml::jni::ScopedJavaLocalRef<jbyteArray> data_array(
-        env, env->NewByteArray(data->GetSize()));
-    env->SetByteArrayRegion(data_array.obj(), 0, data->GetSize(),
-                            reinterpret_cast<const jbyte*>(data->GetMapping()));
+    fml::jni::ScopedJavaLocalRef<jobject> data_array(
+        env, env->NewDirectByteBuffer(const_cast<uint8_t*>(data->GetMapping()),
```

Comment:
Ah yea, this will hit `encodeMessage`.  I'll have to update BinaryCodec to potentially copy that result too.  I'll update it, thanks.

Reply:
Wait no, this only affects Java->Dart replies.  So it would be `decodeMessage` https://github.com/gaaclarke/engine/blob/platform-channels-direct-buffer/shell/platform/android/io/flutter/plugin/common/BasicMessageChannel.java#L173:L173

Reply:
Done, we have to decide if we want to keep `wantsDirectByteBufferForDecoding` though.  If we do I'd have to selectively give a direct byte buffer in this case.  Right now I'm always giving one.

Reply:
Resolved since we want to get rid of wantsDirectByteBufferForDecoding